Cognite.Simulator.Tests/UtilsTests/AutomationUtilsUnitTest.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+
+using Cognite.Simulator.Utils.Automation;
+
+using Microsoft.Extensions.Logging;
+
+using Moq;
+using Moq.Protected;
+
+using Xunit;
+
+using static Cognite.Simulator.Tests.UtilsTests.TestUtilities;
+
+namespace Cognite.Simulator.Tests.UtilsTests
+{
+    [Collection(nameof(SequentialTestCollection))]
+    public class AutomationUtilsUnitTest
+    {
+        private readonly Mock<ILogger<AutomationClient>> _mockLogger;
+        private readonly AutomationConfig _config;
+
+        public AutomationUtilsUnitTest()
+        {
+            _mockLogger = new Mock<ILogger<AutomationClient>>();
+            _config = new AutomationConfig { ProgramId = "Test.Program" };
+        }
+
+        private Mock<AutomationClient> CreateMockClient(bool preShutdownThrows = false, string programId = "Test.Program")
+        {
+            AutomationConfig config = new AutomationConfig { ProgramId = programId };
+            var mock = new Mock<AutomationClient>(_mockLogger.Object, config) { CallBase = true };
+
+            if (preShutdownThrows)
+            {
+                mock.Protected()
+                    .Setup("PreShutdown")
+                    .Throws(new InvalidOperationException("PreShutdown failed"));
+            }
```

Comment:
instead of this `if` here, just do this in `Shutdown_WhenPreShutdownThrows_ExecutesFinallyBlock` on the mock object returned.